### PR TITLE
CT 116 - Only display approved VET TEC providers in the Search Results #19291

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19291
As a Comparison Tool user, I should only see approved VET TEC providers in the search results so that I can choose a provider where I can use my benefits.

## Assumptions:
1. This work is completed behind a flag so that it is only available in the staging environment and below.
2. The "Applicable Law Code" of "VET TEC Only" will be received in the GIDS Weams.csv file in the future.

## Acceptance Criteria:
- [ ] VET TEC providers are only displayed in the "Search Results" page if ALL of the following criteria are met:
- "POO Status" = "APRVD"
- "Applicable Law Code" = "VET TEC ONLY"
- "Non-College Degree Indicator" = "Yes"

## Supporting Artifacts:
1. N/A

## Testing
- Pull branch locally
- Upload weam.csv into local GIDS
- Generate New Preview
- Publish
- Go to http://localhost:3001/gi-bill-comparison-tool/search?vetTecProvider=true

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
Passes QA

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/backend-review-group Could I get a code review on this at your earliest convenience. Thank you!

File: app/models/weam.rb [hunk 1]
```diff
@@ -155,9 +160,14 @@ def approved?
     return false if poo_status.blank? || applicable_law_code.blank?
```
Comment (jholton) [inline app/models/weam.rb]: Let's clean up these `returns` at the end here.  It's a bit hard to follow:
```suggestion
    return false if applicable_law_code.blank? || applicable_law_code =~ Regexp.new("#{ALC1}|#{ALC2}", 'i')
    return false unless poo_status =~ Regexp.new('aprvd', 'i')
    return vet_tec_flags_for_approved? if applicable_law_code =~ Regexp.new("#{ALCVT}", 'i')
    flags_for_approved?
```
We don't need to check if `poo_status.blank?` because the next check (`poo_status =~ Regexp.new('aprvd', 'i')`) covers it.
Please double check my logic!

Comment (zurbergram) [reply app/models/weam.rb]: I think it's ok to leave the returns the way they are as this is more explicit and easier to read for future developers even if they could be cleaned up

Comment (zurbergram) [reply app/models/weam.rb]: made a minor tweak to use a single line if statement for VET TEC ONLY checks

**Comment by jholton** [comment]
Can you update `weam_spec` to cover the `vet_tec_flags_for_approved` scenario?

**Comment by zurbergram** [comment]
> Can you update `weam_spec` to cover the `vet_tec_flags_for_approved` scenario?

added test

**Review by jholton — APPROVED** [review]
LGTM
